sensing/autoware_pointcloud_preprocessor/test/test_distortion_corrector_node.cpp

```diff
@@ -170,10 +171,71 @@ class DistortionCorrectorTest : public ::testing::Test
+    AngleCoordinateSystem vendor)
+  {
+    // Generate all combinations of signs { -, 0, + } x { -, 0, + } for x and y.
+    // Also include the case of (0, 0 ,0)
+    std::vector<Eigen::Vector3f> default_points = {{
+      Eigen::Vector3f(0.0f, 0.0f, 0.0f),    // point 1
+      Eigen::Vector3f(0.0f, 0.0f, 0.0f),    // point 2
+      Eigen::Vector3f(10.0f, 0.0f, 1.0f),   // point 3
+      Eigen::Vector3f(5.0f, -5.0f, 2.0f),   // point 4
+      Eigen::Vector3f(0.0f, -10.0f, 3.0f),  // point 5
+      Eigen::Vector3f(-5.0f, -5.0f, 4.0f),  // point 6
+      Eigen::Vector3f(-10.0f, 0.0f, 5.0f),  // point 7
+      Eigen::Vector3f(-5.0f, 5.0f, -5.0f),  // point 8
+      Eigen::Vector3f(0.0f, 10.0f, -4.0f),  // point 9
+      Eigen::Vector3f(5.0f, 5.0f, -3.0f),   // point 10
+    }};
+
+    std::vector<float> default_azimuths;
+    for (const auto & point : default_points) {
+      if (vendor == AngleCoordinateSystem::VELODYNE) {
+        // velodyne coordinates: x-axis is 0 degrees, y-axis is 270 degrees, angle increase in
+        // clockwise direction
+        float cartesian_deg = std::atan2(point.y(), point.x()) * 180 / autoware::universe_utils::pi;
+        if (cartesian_deg < 0) cartesian_deg += 360;
+        float velodyne_deg = 360 - cartesian_deg;
+        if (velodyne_deg == 360) velodyne_deg = 0;
+        default_azimuths.push_back(velodyne_deg * autoware::universe_utils::pi / 180);
+      } else if (vendor == AngleCoordinateSystem::HESAI) {
+        // hesai coordinates: y-axis is 0 degrees, x-axis is 90 degrees, angle increase in clockwise
+        // direction
+        float cartesian_deg = std::atan2(point.y(), point.x()) * 180 / autoware::universe_utils::pi;
+        if (cartesian_deg < 0) cartesian_deg += 360;
+        float hesai_deg = 90 - cartesian_deg < 0 ? 90 - cartesian_deg + 360 : 90 - cartesian_deg;
+        if (hesai_deg == 360) hesai_deg = 0;
+        default_azimuths.push_back(hesai_deg * autoware::universe_utils::pi / 180);
+      } else if (vendor == AngleCoordinateSystem::CARTESIAN) {
+        // Cartesian coordinates: x-axis is 0 degrees, y-axis is 90 degrees, angle increase in
+        // counterclockwise direction
+        default_azimuths.push_back(std::atan2(point.y(), point.x()));
+      } else {
+        throw std::runtime_error("Invalid angle coordinate system");
+      }
+    }
+
+    return std::make_tuple(default_points, default_azimuths);
+  }
+
+  sensor_msgs::msg::PointCloud2 generateEmptyPointCloudMsg(rclcpp::Time stamp)
+  {
+    sensor_msgs::msg::PointCloud2 empty_pointcloud_msg;
+    empty_pointcloud_msg.header.stamp = stamp;
+    empty_pointcloud_msg.header.frame_id = "lidar_top";
+    empty_pointcloud_msg.height = 1;
+    empty_pointcloud_msg.is_dense = true;
+    empty_pointcloud_msg.is_bigendian = false;
+    empty_pointcloud_msg.width = 0;
+    empty_pointcloud_msg.row_step = 0;
+
+    return empty_pointcloud_msg;
+  }
+
```

Comment:
I think calling `generatePointCloudMsg(true, stamp, {}, {});` will result in the same empty pointcloud, so this function will not be needed. As it is now, the ~10 lines setting the header fields are duplicated.

Reply:
thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/320e9610d1dd43eeffc04bdfb3accd17009bd97a